JASP-Engine/JASP/R/manova.R

```diff
@@ -0,0 +1,193 @@
+  randomFactors <- unlist(options$fixedFactors)
+  
+  datasetDep <- as.matrix(dataset[.v(dependentVariables)])
+  datasetFac <- as.matrix(dataset[.v(randomFactors)])
+  
+  # Main Manova test result for tables
+  manovaModel <- stats::manova(datasetDep ~ datasetFac)
+  
+  allTests <- c("Pillai", "Wilks", "Hotelling-Lawley", "Roy") 
+  whichTests <- allTests[c(options$testPillai, 
+                           options$testWilks, 
+                           options$testHotellingLawley, 
+                           options$testRoy)]
+
+  for (thisTestType in whichTests) {
+    
+    manovaSummary <- summary(manovaModel, 
+                             test = thisTestType, 
+                             intercept = options$includeIntercept)
+    
+    for (case in manovaSummary[[1]][1]) {
+      
+      thisRow <- manovaSummary$stats[case, ]
+      p <- thisRow['Pr(>F)']
+      
+      # Add results for each test type to results object
+      results[["manova"]][[thisTestType]][[case]] <- list(
+        testType = thisTestType,
+        cases      = case,
+        testStat = thisRow[thisTestType],
+        df         = thisRow['Df'],
+        appF        = thisRow['approx F'],
+        dfNum         = thisRow['num Df'],
+        dfDen    = thisRow['den Df'],
+        p             = p,
+        VovkSellkeMPR = .VovkSellkeMPR(p)
+      )
+    }
+  }
+
+
+  
+  # Save results to state
+  jaspResults[["stateManovaResults"]] <- createJaspState(results)
+  # jaspResults[["stateManovaResults"]]$dependOn(
+  #   c("dependent", "fixedFactors", "testPillai", "testWilks", 
+  #     "testHotellingLawley", "testRoy", "includeIntercept",
+  #     "VovkSellkeMPR")
+  # )
+  
+  # Return results object
+  return(results)
+}
+
+.manovaTableMain <- function(jaspResults, dataset, options, manovaResults, errors) {
+  if (!is.null(jaspResults[["manovaTable"]])) return()
+  
+  # Create table
+  manovaTable <- createJaspTable(title = "MANOVA")
+  # manovaTable$dependOnOptions <- c("dependent", "fixedFactors", "testPillai", "testWilks", 
+                         # "testHotellingLawley", "testRoy", "includeIntercept",
+                         # "VovkSellkeMPR")
+  
+
+  # Add columns to table
+  manovaTable$addColumnInfo(name = "cases",   title = "Cases", type = "string", combine = TRUE)
+  manovaTable$addColumnInfo(name = "df",      title = "df",      type = "integer")
+  manovaTable$addColumnInfo(name = "appF",    title = "Approx. F",      type = "number")
+  manovaTable$addColumnInfo(name = "testStat",title = "Wilks",     type = "number")
+  manovaTable$addColumnInfo(name = "dfNum",   title = "Num df",      type = "integer")
+  manovaTable$addColumnInfo(name = "dfDen",   title = "Den df",      type = "integer")
+  manovaTable$addColumnInfo(name = "p",       title = "p",          type = "pvalue")
+  
+  if (options$VovkSellkeMPR) {
+    manovaTable$addColumnInfo(name = "VovkSellkeMPR", title = "VS-MPR\u002A", type = "number")
```

Comment:
`showspecifiedcolumnsonly` in jaspTable otherwise you shouldn't fill the table with everything by default.